library/std/tests/sync/mpsc.rs

```diff
@@ -438,18 +438,23 @@ fn stress_recv_timeout_two_threads() {
     });
 
     let mut recv_count = 0;
+    let mut got_timeout = false;
     loop {
         match rx.recv_timeout(timeout) {
             Ok(n) => {
                 assert_eq!(n, 1usize);
                 recv_count += 1;
             }
-            Err(RecvTimeoutError::Timeout) => continue,
+            Err(RecvTimeoutError::Timeout) => {
+                got_timeout = true;
+                continue;
+            }
             Err(RecvTimeoutError::Disconnected) => break,
         }
     }
 
     assert_eq!(recv_count, stress);
+    assert!(got_timeout);
```

Comment:
the `stress_factor()` will be 1 by default, which will give us `stress == 51`, so we spawn 51 threads, send 51 things, and assert that we receive all 51 things before we encounter the "sender hung up on us" message.

the timeout factor will enforce that about ~half of these will be beyond the timeout boundary, but we can't rely on that because we will eventually reach the "leading edge" of timeouts, and from there, because of buffering for messages, we will drain without encountering a timeout even once.

which does present a hazard, and I started walking through the time boundary, but then the end of the spawn loop still gives us the full 10 milliseconds.

hmm...

yeah I *think* receiving 50 messages will go faster than ~9.9 milliseconds

Reply:
> the `stress_factor()` will be 1 by default, which will give us `stress == 51`, so we spawn 51 threads, send 51 things, and assert that we receive all 51 things before we encounter the "sender hung up on us" message.

The test doesn't spawn `stress` threads; it only spawns *one* thread, which sends `stress` messages, half of which it sleeps 2x the timeout before sending.

Reply:
hmm.

Reply:
oh yeah you're right.